test/studies/ml/lib/Chai.chpl

```diff
@@ -436,27 +431,27 @@ module Chai {
 
         proc init() { }
 
-        proc forwardProp(input: Tensor(?inRank)): Tensor(1) do
+        proc ref forwardProp(input: Tensor(?inRank)): Tensor(1) do
             return input.flatten();
 
-        proc forwardPropBatch(batch: [] ?tensorType): [] Tensor(1) where isSubtype(tensorType,Tensor) do
-            return [b in batch] forwardProp(b);
+        proc ref forwardPropBatch(batch: [] ?tensorType): [] Tensor(1) where isSubtype(tensorType,Tensor) do
+            return forwardProp(batch);
 
-        proc backward(delta: Tensor(1), input: Tensor(?inRank)): Tensor(inRank) do
+        proc ref backward(delta: Tensor(1), input: Tensor(?inRank)): Tensor(inRank) do
             return delta.reshape(input.domain);
 
-        proc backwardBatch(deltas: [] Tensor(1), inputs: [] ?tensorType): [] tensorType where isSubtype(tensorType,Tensor) do
-            return [(d,i) in zip(deltas,inputs)] backward(d,i);
+        proc ref backwardBatch(deltas: [] Tensor(1), inputs: [] ?tensorType): [] tensorType where isSubtype(tensorType,Tensor) do
+            return backward(deltas,inputs);
```

Comment:
Oh, I'm now realizing that this may be just using promotion?

Reply:
Yes